### PR TITLE
Refetch user session when it expires

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]
approving code changes

File: app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx [hunk 1]
```diff
@@ -44,12 +46,17 @@ export interface SessionProviderProps {
   readonly mainPageUrl: URL
   readonly registerAuthEventListener: listen.ListenFunction | null
   readonly userSession: (() => Promise<cognito.UserSession | null>) | null
+  readonly refreshUserSession: (() => Promise<unknown>) | null
   readonly children: React.ReactNode
 }
 
+const FIVE_MINUTES = 300_000
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx]: our convention is to include units. so `FIVE_MINUTES_MS` and `SIX_HOURS_MS`

File: app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx [hunk 2]
```diff
@@ -44,12 +46,17 @@ export interface SessionProviderProps {
   readonly mainPageUrl: URL
   readonly registerAuthEventListener: listen.ListenFunction | null
   readonly userSession: (() => Promise<cognito.UserSession | null>) | null
+  readonly refreshUserSession: (() => Promise<unknown>) | null
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx]: not sure if a return value of `Promise<unknown>` is particularly useful...

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx]: I used unknown here bc from the component perspective it doesn't matter if the promise returns smth or not.
I didn't use `Promise<void>` here bc if a callback, passed in `refreshUserSession` returns `Promise<Smth>` ts with throw an error. Is there a better solution for that case?

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/providers/SessionProvider.tsx]: wrapping the body in braces is fine in that situation imo - if the return value isn't supposed to be used, then it shouldn't be returned at runtime either

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
ready to merge as soon as all CI are fixed

**Pull request merged**